mobile-app/app/screens/AppNavigator/screens/Dex/hook/SwappableTokens.ts

```diff
@@ -0,0 +1,95 @@
+    const tokens = useSelector((state: RootState) => tokensSelector(state.wallet))
+
+    const _allTokens = useMemo(() => {
+        return poolpairs.reduce((tokensInPair: TokenState[], pair: DexItem): TokenState[] => {
+            const hasTokenA = tokensInPair.some(token => pair.data.tokenA.id === token.id)
+            const hasTokenB = tokensInPair.some(token => pair.data.tokenB.id === token.id)
+            const tokensToAdd: TokenState[] = []
+            if (!hasTokenA) {
+                tokensToAdd.push(pair.data.tokenA)
+            }
+            if (!hasTokenB) {
+                tokensToAdd.push(pair.data.tokenB)
+            }
+
+            return [...tokensInPair, ...tokensToAdd]
+        }, [])
+    }, [poolpairs])
+
+    const toTokens = useMemo(() => {
+        const swappableToTokens = swappableTokens[fromTokenId === '0_unified' ? '0' : fromTokenId ?? '']
+        if (swappableToTokens === undefined) {
+            return []
+        }
+
+        const toTokens: BottomSheetToken[] = swappableToTokens.swappableTokens
+            .map((token) => {
+                const tokenId = token.id === '0' ? '0_unified' : token.id
+                const tokenData = _allTokens.find(t => t.id === token.id)
+
+                return {
+                    tokenId: tokenId,
+                    available: new BigNumber(tokenData?.reserve ?? NaN),
+                    token: {
+                        displaySymbol: token.displaySymbol,
+                        name: '', // not available in API,
+                        symbol: token.symbol
+                    },
+                    reserve: tokenData?.reserve ?? '' // TODO(PIERRE): Ask whale to add reserve on response
+                }
+            })
+
+        return toTokens
+    }, [swappableTokens, fromTokenId])
+
+    const fromTokens = useMemo(() => {
+        const swappableFromTokens: BottomSheetToken[] = _allTokens
+            .map((token) => {
+                const tokenId = token.id === '0' ? '0_unified' : token.id
+                const ownedToken = tokens.find(t => t.id === tokenId)
+                return {
+                    tokenId: tokenId,
+                    available: new BigNumber(ownedToken === undefined ? 0 : ownedToken.amount),
+                    token: {
+                        displaySymbol: token.displaySymbol,
+                        name: '', // not available in API,
+                        symbol: token.symbol
+                    },
+                    reserve: token.reserve
+                }
+            }).sort((a, b) => b.available.minus(a.available).toNumber())
+
+        return swappableFromTokens
+    }, [poolpairs])
+
+    useEffect(() => {
+        if (fromTokenId !== undefined) {
+            dispatch(fetchSwappableTokens({
+                client, fromTokenId: fromTokenId === '0_unified' ? '0' : fromTokenId
+            }))
+        }
+    }, [blockCount, fromTokenId])
```

Comment:
Can we implement cashing mechanism here to avoid repetitive api calls when user reselect the token?
for eg: If user first select BTC as from token then change it to ETH and then again resect BTC (no new block get mined in this switching).
In this case we can save one api call.
cc @thedoublejay

Reply:
Yep, we can cache this definitely. By symbol/block

Reply:
Actually, I think we don't need to call this on every block update. Swappable tokens rarely updates (usually when there are only new tokens).

Reply:
Updated. Removed blockCount on `dep array` to reduce rerenders, but kept the blockCount in the caching